core/src/main/java/feast/core/validators/FeatureSetValidator.java

```diff
@@ -44,8 +44,8 @@ public static void validateSpec(FeatureSet featureSet) {
       throw new IllegalArgumentException("Feature set label keys must not be empty");
     }
 
-    checkValidCharacters(featureSet.getSpec().getProject(), "project");
-    checkValidCharacters(featureSet.getSpec().getName(), "name");
+    checkValidCharacters(featureSet.getSpec().getProject(), "project::name");
+    checkValidCharacters(featureSet.getSpec().getName(), "featureset::name");
```

Comment:
Would it make more sense to have a new argument like `resource` and `field` instead of `::`

Reply:
I added resource. I think it's clearer to show what they inputted as well, so I replaced field with the actual input itself in the error message.